### PR TITLE
feat: Red Hat product processor and related refactoring

### PR DESCRIPTION
**Pull request by goldmann**
* allow for multiple proccessors to be executed for a single CLI run
* NCL-7791: created a temporary mapping that would help to populate the SBOM with the product-related information until we have it in PNC
* NCL-7794: implemented the product processor (`redhat-product`)
* refactoring of the processor codebase so that above changes are possible
* introduced mixin so that we can share same options
* some class renames
* introduced a different way to run CLI tests which should be more used in the future tests

### CODE REVIEW

File: cli/src/main/java/org/jboss/sbomer/cli/commands/processor/RedHatProductProcessCommand.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        }
+
+        BuildConfiguration buildConfig = pncService.getBuildConfig(build.getBuildConfigRevision().getId());
+
+        if (buildConfig == null) {
+            throw new ApplicationException(
+                    "BuildConfig related to the SBOM could not be found in PNC, interrupting processing");
+        }
+
+        ProductVersionMapping mapping = productVersionMapper.getMapping().get(buildConfig.getProductVersion().getId());
```
Comment (vibe13) [inline cli/src/main/java/org/jboss/sbomer/cli/commands/processor/RedHatProductProcessCommand.java:97]: ProductVersion might be null in the BuildConfigs

**Review by vibe13 — CHANGES_REQUESTED** [review]
Great! Just a minor comment about a possible NPE

File: cli/src/main/java/org/jboss/sbomer/cli/commands/processor/RedHatProductProcessCommand.java (hunk 1 above)
Comment (goldmann) [reply cli/src/main/java/org/jboss/sbomer/cli/commands/processor/RedHatProductProcessCommand.java:97]: Thanks, will update it!

Comment (goldmann) [reply cli/src/main/java/org/jboss/sbomer/cli/commands/processor/RedHatProductProcessCommand.java:102]: This should be now covered!

**Review by vibe13 — APPROVED** [review]

**Pull request merged**